From patchwork Wed Apr 17 05:18:11 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Nadav Amit <nadav.amit@gmail.com>
X-Patchwork-Id: 10904459
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 9DBA4922
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 17 Apr 2019 05:18:22 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 7BAB7285A6
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 17 Apr 2019 05:18:22 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 6E2D92877B; Wed, 17 Apr 2019 05:18:22 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-8.0 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,FREEMAIL_FROM,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI
	autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 206D6285A6
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 17 Apr 2019 05:18:21 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1726395AbfDQFSU (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 17 Apr 2019 01:18:20 -0400
Received: from mail-pg1-f194.google.com ([209.85.215.194]:44075 "EHLO
        mail-pg1-f194.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1725767AbfDQFSU (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 17 Apr 2019 01:18:20 -0400
Received: by mail-pg1-f194.google.com with SMTP id i2so11427328pgj.11
        for <kvm@vger.kernel.org>; Tue, 16 Apr 2019 22:18:20 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20161025;
        h=from:to:cc:subject:date:message-id:mime-version
         :content-transfer-encoding;
        bh=BTyDElIJ2E7CwHvVHsBeed3CrwPmyHiw4Vz7y7oRRtc=;
        b=GLOGjgft6cvzU+6IOQrxUfUDWUvyrDUllqUoKXkCRyrEaWQotg5RLVgybbFNTKU4Jn
         WeHSjgtt5Q1RVyRFExUcoH1U3q5Hv2TMfE3O57F5T+kR7srRZRsMIXX++fQrBitP15hu
         bjXEvYn3qbQwQC11bidsXb90ioYMo6Jjc4N5F36nTe4wCwMB1ymAfTs2Kv1pxfxxhINQ
         Pmkv9iWju0320aRC0sD/nxgI3NYQUVTzzelMw9RjZ0sAdZ2OiPKH/y5GhekBDjeldfkV
         N5PoA1Vs6P6QBq6vjf4y28K50hMZ5+iXJ27pg0F4guNMm7nyglKK67SUNqN3aNVTEVsC
         KfmA==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:mime-version
         :content-transfer-encoding;
        bh=BTyDElIJ2E7CwHvVHsBeed3CrwPmyHiw4Vz7y7oRRtc=;
        b=mljsS+gTKNDTKXdSZHllaOk/w5gmdWjZUEiAUO0kEPhg0A4tYpN6YpxIus+BAXM1d2
         ZtmuFvM2+OWeSR1kqoXV//WOx+c1dNsozt3IwrhDYEXFqvcAu7rIkXsghfXWWNBrSPI8
         n5HIIzgnw1Us431/Y5HPF9pwY18GvMRkhjbsJEm6q65hvNtLJ29XkT79hhq2WLvGglxH
         dsaGuQbInKHTjsA1rAzBRlWTIu53he8tOO/SvA6Us87a8MTHZvCiz8Hg+fD2PYrnsKan
         idltALxr3JOy8UJgTtQGAI+kGVVWfUxcEiqHmiSRkWguP6ILBK/HWg2IOjzn4qE6W86H
         ySdg==
X-Gm-Message-State: APjAAAVwd4R5qpb9FQOXm51d0WfftBEZKlPfeSnEoIHbBh8NXm8RSyrv
        K/tTR9+WWeNh+C2DwXMLucs=
X-Google-Smtp-Source: 
 APXvYqxARJ7obhTDAfXhQuLWFVinjKWhSpXNEupz/gqw8sIMaY0tcjWTt6Yx1ZX21mRtxRFOniOl1w==
X-Received: by 2002:aa7:8282:: with SMTP id s2mr87689172pfm.7.1555478298360;
        Tue, 16 Apr 2019 22:18:18 -0700 (PDT)
Received: from htb-2n-eng-dhcp405.eng.vmware.com ([66.170.99.1])
        by smtp.gmail.com with ESMTPSA id
 m25sm72072573pfa.175.2019.04.16.22.18.17
        (version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
        Tue, 16 Apr 2019 22:18:17 -0700 (PDT)
From: nadav.amit@gmail.com
To: Paolo Bonzini <pbonzini@redhat.com>
Cc: kvm@vger.kernel.org,
        Sean Christopherson <sean.j.christopherson@intel.com>,
        Nadav Amit <nadav.amit@gmail.com>
Subject: [kvm-unit-tests PATCH] lib: Do not report failures when test passes
Date: Tue, 16 Apr 2019 22:18:11 -0700
Message-Id: <20190417051811.18375-1-nadav.amit@gmail.com>
X-Mailer: git-send-email 2.19.1
MIME-Version: 1.0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

From: Nadav Amit <nadav.amit@gmail.com>

Currently, if a test is expected to fail, but surprisingly it passes,
the test is considered as "failing". This means that running on old
KVM-unit-tests on new KVM versions can falsely report failures.

Fix it and simplify the logic.

Signed-off-by: Nadav Amit <nadav.amit@gmail.com>
---
 lib/report.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/lib/report.c b/lib/report.c
index ca9b4fd..8c8db14 100644
--- a/lib/report.c
+++ b/lib/report.c
@@ -96,10 +96,12 @@ static void va_report(const char *msg_fmt,
 	puts("\n");
 	if (skip)
 		skipped++;
-	else if (xfail && !pass)
-		xfailures++;
-	else if (xfail || !pass)
-		failures++;
+	else if (!pass) {
+		if (xfail)
+			xfailures++;
+		else
+			failures++;
+	}
 
 	spin_unlock(&lock);
 }
